app/org/sagebionetworks/bridge/models/schedules/ActivityScheduler.java

```diff
@@ -127,8 +128,12 @@ protected DateTime getFirstEventDateTime(ScheduleContext context, String eventId
      * If scheduling hasn't reached the end time, or hasn't accumulated the minimum number of tasks, returns true, or 
      * false otherwise. 
      */
-    protected boolean shouldContinueScheduling(ScheduleContext context, DateTime scheduledTime, List<ScheduledActivity> scheduledActivities) {
-        boolean boundaryNotMet = scheduledTime.isBefore(context.getEndsOn()) || hasNotMetMinimumCount(context, scheduledActivities.size());
+    protected boolean shouldContinueScheduling(ScheduleContext context, DateTime scheduledTime,
+            List<ScheduledActivity> scheduledActivities) {
+        
+        boolean boundaryNotMet = scheduledTime.withZoneRetainFields(context.getZone()).isBefore(context.getEndsOn()) || 
```

Comment:
Changed this. One test broke, but it was added as part of this check in and the results look correct to me (the test was wrong).